src/geometry.rs

```diff
@@ -57,3 +153,25 @@ where
         Self::from((cmp::max(tuple.0, other.0), cmp::max(tuple.1, other.1)))
     }
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    #[test]
+    fn parse_matrix() {
+        let matrix = "1, 0, 0; 0, 1, 0; 0, 0, 1";
+        assert_eq!(
+            vec![1, 0, 0, 0, 1, 0, 0, 0, 1],
+            Matrix3x3::<i32>::from_str(matrix).unwrap().storage
+        );
+
+        let error =
+            String::from("Mismatched size when createing Matrix3x3 from Vec, expected 9 got 3");
+        let matrix = "1, 0, 0, 0, 1, 0, 0, 0, 1";
+        assert_eq!(&error, &Matrix3x3::<i32>::from_str(matrix).err().unwrap().to_string());
+
+        let matrix = "1; 0; 0; 0; 1; 0; 0; 0; 1;";
+        assert_eq!(&error, &Matrix3x3::<i32>::from_str(matrix).err().unwrap().to_string());
```

Comment:
I don't really mind either way, but honestly checking for the exact error is more likely to break the test than bring any benefits. Just checking `is_err()` should be sufficient.